doc/concepts.md

```diff
@@ -1,190 +1,267 @@
-    with the same header as DLiteInstance, but includes more fields
-    needed to describe their instances.  Entities are a special case
-    of metadata, whos instances are the actual data.
-
-    All metadata is immutable.
+Currently DLite does not enforce that user-defined URIs must follow
+the [RFC 3986] standard for a [valid URI], but it is recommended to do
+so in order to allow using the URI as a valid [RDF]
+subject or object in a knowledge base.
+For this purpose, DLite also allow to refer to data instances using
+id's of the form `<metadata_uri>/<uuid>` (for example
+http://onto-ns.com/meta/0.1/Collection/db6e092b-20f9-44c1-831b-bd597c96daae).
 
 
 Metadata semantics
 ------------------
-The semantics used to by any type of metadata to describe its instances
-contains three elements:
-
-  - dimensions
-  - properties
-  - relations
-
-The three first properties of all metadata schemas (metadata who's
-instances are metadata) must be "dimensions", "properties" and
-"relations" in this order.  However, it is possible to omit
-"relations" if the metadata instance has no other properties.
-
+![The datamodel of DLite.](figs/datamodel.svg)
+
+_**Figure 3**. The DLite data model._
+
+The DLite data model is defined by the [Datamodel ontology] and shown
+schematically in figure 3.
+To the left we have the same hierarchy as shown in Figure 2.
+Then we have *instance* and *metadata* as more general concepts.
+
+As discussed above, all instances have an UUID that is used as their
+main identifier for database lookup.
+All metadata as well as some data instances can also be identified by
+an URI (in which case the UUID is a hash of this URI).
+Instances also have a *meta* field that refer to the metadata that it
+is described by (or that it is an instance of).
+
+In general metadata describe their instances using *dimensions*,
+*properties* and *relations*.
+Properties describe data in terms of keyword-value pairs, dimensions
+enable efficient description of multi-dimensional arrays and relations
+can describe anything that can be represented in a knowledge base.
+Together they provide a general mean to describe all types of data
+that can be represented digitally.
+Note however, that not all metadata uses all of these three ways to
+describe their instances.
+For example has entities only dimensions and properties.
+
+
+### Dimension
+A metadata dimension simply provide a *name* and a human *description* of a
+given dimension of an array property.
+
+
+### Property
+A property describe an element or item of an instance and has the following
+attributes:
+- *name*: a name identifying the property.
+- *type*: the type of the described property, f.ex. an integer.
+- *$ref*: formally a part of type.
+  `$ref` is used together with the "ref" type, which is a special datatype for
+  referring to other instances.
+- *shape*: The dimensions of a multi-dimensional properties.
+  This is a list of dimension names referring to the dimensions defined above.
```

Comment:
I think this is a bit confusing it is a list of what? Dimensions are numbers, no`?